internal/cluster/cluster.go

```diff
@@ -1039,6 +1040,88 @@ func (m *Manager) IsReadyForInstallation(c *common.Cluster) (bool, string) {
 	return true, ""
 }
 
+func (m *Manager) detectAndStoreCollidingIPsForCluster(cluster *common.Cluster, db *gorm.DB) error {
+	if db == nil {
+		db = m.db
+	}
+	// We want to calculate ip collisions only when in pre-install states since it is needed for pre-install validations
+	allowedStates := []string{
+		models.ClusterStatusPendingForInput,
+		models.ClusterStatusInsufficient,
+
+		models.ClusterStatusReady,
+	}
+	if !funk.ContainsString(allowedStates, swag.StringValue(cluster.Status)) {
+		return nil
+	}
+
+	hosts := cluster.Hosts
+	sort.Slice(hosts, func(i, j int) bool {
+		return hosts[i].ID.String() < hosts[j].ID.String()
+	})
+	ipCollisions := make(map[string][]string)
+	collidingIPSWithMacs := make(map[string][]string)
+	for _, host := range hosts {
+		if len(host.Connectivity) > 0 {
+			connectivityReport, err := hostutil.UnmarshalConnectivityReport(host.Connectivity)
```

Comment:
I'm not really sure where this cache would be implemented, has this been done elsewhere in the code?
Surely this would depend on us knowing the previous checksum, I am not sure this is the situation here?